### DISCUSSION TITLE
nextjs with next-routes upgrade

### DISCUSSION THREAD

**Discussion by eliransu** [vercel/next.js#20361] [Help]
Hi, 

as of now, we use nextjs "8.1.1-canary.0" version with next-routes "1.4.2" for next routing.
we MUST upgrade next from 8 to 10 (super important for us) but we don't know even how to start (8 && next-routes -> 10)

maybe someone can help me to understand what's the best way to achieve it?

Thanks in advance!

**Comment by M1ck0** [comment]
You have guide [here](https://nextjs.org/docs/upgrading), and [this](https://nextjs.org/docs/advanced-features/codemods) will also help you.

All breaking changes are listed and it shouldn't be too hard to upgrade to the Next 10.

I do suggest making a copy of the current project since you never know if something can go wrong.

Upgrade to Next 9 first. If that is successful then upgrading to Next 10 will be as easy as `npm install next@latest`, since there is no breaking changes between `9` and `10`
